### ISSUE TITLE
Add Pressure Stall Information (PSI) for cgroups

### ISSUE THREAD

**Issue by rbjorklin** [netdata/netdata#10134]
##### Feature idea summary
Pressure Stall Information was originally requested in #5901 and implemented in the `proc.plugin` in #7209. This request is for adding PSI to the [cgroups.plugin](https://learn.netdata.cloud/docs/agent/collectors/cgroups.plugin/). The information is available under `/sys/fs/cgroup/unified/*/*/[io|cpu|memory].pressure`

##### Expected behavior
Each container should have it's own graphs for PSI similar to how it works for the `proc.plugin`.

**Comment by hexchain** [comment]
I think these are already in place since https://github.com/netdata/netdata/commit/8a70725c132deb6fc313edcdccaea756a685fc49?

**Comment by vlvkobal** [comment]
Yes. It should work right out of the box.
```
[plugin:cgroups]
	# enable cpu pressure = auto
	# enable io some pressure = auto
	# enable io full pressure = auto
	# enable memory some pressure = auto
	# enable memory full pressure = auto
```

**Comment by rbjorklin** [comment]
Then maybe this issue should have been reported as a bug instead. I can see PSI on the host level but not on the container level using Netdata 1.21.1.

**Comment by vlvkobal** [comment]
What is the value of `path to unified cgroups` in the `[plugin:cgroups]` section of `yourhost:19999/netdata.conf`?

Please also provide the full path for the `cpu.pressure` file for any container in your hierarchy.

**Comment by vlvkobal** [comment]
It looks like you have both versions of cgroups enabled. Please set `use unified cgroups = yes`.

**Comment by rbjorklin** [comment]
This does indeed work when you [RTFM](https://learn.netdata.cloud/docs/agent/collectors/cgroups.plugin/#unified-cgroups-cgroups-v2-support). Sorry guys and thanks for the pointer!

**Comment by rbjorklin** [comment]
Actually when enabling the unified cgroups all the other metrics disappear. Is this intended?

**Comment by ilyam8** [comment]
@rbjorklin are you sure you have expected cgroups `/sys/fs/cgroup/unified/**`?

I created docker container and i see it

```cmd
# tree -d 1 /sys/fs/cgroup/cpu
/sys/fs/cgroup/cpu
├── docker
│   └── b3a14244e8bc0fc3ee6d3a9632e8a3d08dd274a96891ac8d1153dd607af37eef
```

I can't find it in `/sys/fs/cgroup/unified/`.

**Comment by rbjorklin** [comment]
I'm not entirely sure what's expected but this is what I see:

```
$ sudo tree -d 1 /sys/fs/cgroup/cpu
1 [error opening dir]
/sys/fs/cgroup/cpu
|-- lxc
|   |-- one-11370
|   |-- one-11372
|   |-- one-11412
|   |-- one-11428
|   |-- one-11965
|   |-- one-4914
|   |-- one-6642
|   |-- one-7314
|   |-- one-7690
|   |-- one-9477
|   |-- one-9480
|   `-- one-9974
...
```

and this is for the `unified` directory:
```
$ sudo tree -d 1 /sys/fs/cgroup/unified/
1 [error opening dir]
/sys/fs/cgroup/unified/
|-- init.scope
|-- lxc
|   |-- one-11370
|   |-- one-11372
|   |-- one-11412
|   |-- one-11428
|   |-- one-11965
|   |-- one-4914
|   |-- one-6642
|   |-- one-7314
|   |-- one-7690
|   |-- one-9477
|   |-- one-9480
|   `-- one-9974
...
```

**Comment by hexchain** [comment]
Can't say for sure, but I guess some stats do not appear if the corresponding controller on that subtree is not enabled. Could you please check `cgroup.controllers` (for that tree) and `cgroup.subtree_control` (for subtrees) in `/sys/fs/cgroup/unified/lxc` and any of the descendants?

On my machine (I have `systemd.unified_cgroup_hierarchy`), I have `cpu io memory hugetlb pids` at the root and `cpuset cpu io memory hugetlb pids` at the subtree of a podman/crun container. In netdata, I get more than 10 graphs for that container.

**Comment by rbjorklin** [comment]
`cgroup.controllers` and `cgroup.subtree_control` both exist under `/sys/fs/cgroup/unified/lxc` and descendants but they are empty. Is that expected?

The kernel parameter `systemd.unified_cgroup_hierarchy` is not set. Not sure if it matters but the kernel is `5.3.0-40-generic #32~18.04.1-Ubuntu`

**Comment by hexchain** [comment]
Being empty means no controller is enabled. That is probably the reason why you don't have other metrics after enabling unified cgroups in netdata.

**Comment by rbjorklin** [comment]
Hmm... It looks like this was indeed requested in #3750 and later on implemented in #5407. `io.stat` and `memory.stat` among others are indeed missing. I think this can be written of as user error in the end.
